src/Stryker.Core/Stryker.Core/Testing/ProcessExecutor.cs

```diff
@@ -29,24 +34,44 @@ public ProcessExecutor(bool redirectOutput = true)
                 RedirectStandardError = _redirectOutput
             };
 
-            foreach(var environmentVariable in environmentVariables ?? Enumerable.Empty<KeyValuePair<string, string>>())
+            foreach (var environmentVariable in environmentVariables ?? Enumerable.Empty<KeyValuePair<string, string>>())
             {
                 info.EnvironmentVariables[environmentVariable.Key] = environmentVariable.Value;
             }
 
-            var process = Process.Start(info);
-            string output = "";
-            if(_redirectOutput)
+            using (var process = Process.Start(info))
             {
-                output = process.StandardOutput.ReadToEnd();
-            }
-            process.WaitForExit();
+                string output = "";
+                var processDoneHandle = new ManualResetEvent(false);
 
-            return new ProcessResult()
-            {
-                ExitCode = process.ExitCode,
-                Output = output
-            };
+                Task.Run(() =>
```

Comment:
This is a pretty long method as it is. Could we move this code to a `private Task<string> RunProcessAsync(Process p)` method?